mathics/doc/common_doc.py

```diff
@@ -304,10 +304,10 @@ def skip_doc(cls) -> bool:
     return cls.__name__.endswith("Box") or (hasattr(cls, "no_doc") and cls.no_doc)
 
 
-def skip_module_doc(module, modules_seen) -> bool:
+def skip_module_doc(module, must_be_skipped) -> bool:
```

Comment:
There are several reasons to skip a module in the documentation system. However, to skip it because we already visited should not be a good criterion. The way in which we walk the modules should be more predictable to allow testing the process.

Reply:
Yes, I agree. Things were given to us in kind of a mess, and we haven't dug our way out yet.